### ISSUE TITLE
Create Graph from JSON document

### ISSUE THREAD

**Issue by jexp** [neo4j-contrib/neo4j-apoc-procedures#161]
Apply automatic rules for generating the graph (inverse to toTree)

**Comment by jexp** [comment]
Provide a config as JSON that describes the various levels by example (a bit like GraphQL).

```
{ label: "Person"
  attr : ["name","age"],
 friends: { type: "KNOWS", label: "Person",attr : ["name","age"] }
 location: { type: "LIVES_IN>", multi:false, label: "City",attr : ["name","country"] }
}
```

**Comment by brownish** [comment]
Any progress on this? I would absolutely love this ability, especially if it's able to handle deeply nested trees something like this:

```
WITH [
    {
        name: 'Untempered'
    },
    {
        name: 'Indeterminate'
    },
    {
        name: 'Organic',
        subtypes: [
            {
                name: 'Ash'
            },
            {
                name: 'Shell'
            },
            {
                name: 'Bone'
            },
            {
                name: 'Burned Bone'
            }
        ]
    },
    {
        name: 'Inorganic',
        subtypes: [
            {
                name: 'Grit',
                subtypes: [
                    {
                        name: 'Hematite Grit'
                    },
                    {
                        name: 'Limestone Grit'
                    },
                    {
                        name: 'Granitic Grit'
                    }
                ]
            },
            {
                name: 'Grog'
            },
            {
                name: 'Sand'
            },
            {
                name: 'Burned Bone'
            }
        ]
    }
] as temper
```

**Comment by conker84** [comment]
closed via ab5289d
